KubeArmor/core/kubeUpdate.go

```diff
@@ -218,6 +222,20 @@ func (dm *KubeArmorDaemon) UpdateEndPointWithPod(action string, pod tp.K8sPod) {
 		}
 		dm.ContainersLock.Unlock()
 
+		dm.DefaultPosturesLock.Lock()
+		if val, ok := dm.DefaultPostures[newPoint.NamespaceName]; ok {
+			newPoint.DefaultPosture = val
+		} else {
+			globalDefaultPosture := tp.DefaultPosture{
+				FileAction:         cfg.GlobalCfg.DefaultFilePosture,
+				NetworkAction:      cfg.GlobalCfg.DefaultNetworkPosture,
+				CapabilitiesAction: cfg.GlobalCfg.DefaultCapabilitiesPosture,
+			}
+			dm.DefaultPostures[newPoint.NamespaceName] = globalDefaultPosture
```

Comment:
why do we need to update dm.DefaultPostures here?

Reply:
WatchDefaultPosture only updates the values in `dm.DefaultPosture` when there are annotation related updates. When there is an endpoint related update but that namespace has no default posture set due to no annotations, we set the namespace annotation to the global default posture for this endpoint and all future endpoints in this namespace.